src/fsharp/IlxGen.fs

```diff
@@ -7108,9 +7118,11 @@ and GenAbstractBinding cenv eenv tref (vref: ValRef) =
 
         assert witnessInfos.IsEmpty
 
+        let _, retTy = stripFunTy g vref.Type
```

Comment:
I think this will fail for generic methods.  Instead, notice that `GetMethodSpecForMemberVal` is already cracking the `returnTy`, so return `returnTy` from that and collect it 4 lines above.